src/App.jsx

```diff
@@ -1,247 +1,257 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
+import classNames from 'classnames';
+
+import usersFromServer from './api/users';
+import categoriesFromServer from './api/categories';
+import productsFromServer from './api/products';
+
+const products = productsFromServer.map((product) => {
+  const category = categoriesFromServer.find(currentCategory => (
+    product.categoryId === currentCategory.id
+  ));
+  const user = usersFromServer.find(currentUser => (
+    currentUser.id === category.ownerId
+  ));
+
+  return {
+    productId: product.id,
+    productName: product.name,
+    categoryIcon: category.icon,
+    categoryName: category.title,
+    userName: user.name,
+    userSex: user.sex,
+    userId: user.id,
+  };
+});
+
+export const App = () => {
+  const [selectedUserId, setSelectedUserId] = useState(null);
+  const [searchInputValue, setSearchInputValue] = useState('');
+
+  const handleClickUser = (UserId) => {
+    setSelectedUserId(UserId);
+  };
+
+  const inputValueLowerCase = searchInputValue.toLowerCase();
+
+  const filteredProductsByName = products.filter(product => (
+    product.productName.toLowerCase().includes(inputValueLowerCase)
+  ));
+
+  const filteredProductsByUser = filteredProductsByName.filter(product => (
+    product.userId === selectedUserId
+  ));
+
+  const visibleProducts = selectedUserId
+    ? filteredProductsByUser
+    : filteredProductsByName;
+
+  const onReset = () => {
```

Comment:
Why `onReset`? You should read this article one more time: https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364
```suggestion
  const handleResetClick = () => {
```